### PR TITLE
Check the vaa confirmations based on timestamp

### PR DESCRIPTION
**Pull request by Lbqds**

### CODE REVIEW

File: node/pkg/alephium/watcher.go [hunk 1]
```diff
@@ -366,7 +368,8 @@ func (w *Watcher) handleEvents_(
 	pendingEvents := map[string]*UnconfirmedEventsPerBlock{}
 
 	process := func(height int32) error {
-		logger.Debug("processing events", zap.Int32("height", height))
+		now := time.Now().UnixMilli()
+		logger.Debug("processing events", zap.Int32("height", height), zap.Int64("now", now))
```
Comment (h0ngcha0) [inline node/pkg/alephium/watcher.go:374]: [Nit] I guess logger has the current timestamp already

File: node/pkg/alephium/watcher.go [hunk 2]
```diff
@@ -385,16 +388,17 @@ func (w *Watcher) handleEvents_(
...
-				if blockEvents.header.Height+int32(eventConfirmations) > height {
-					logger.Debug(
-						"event not confirmed",
-						zap.String("txId", event.TxId),
-						zap.Int32("blockHeight", blockEvents.header.Height),
-						zap.Uint8("confirmations", eventConfirmations),
-					)
+				consistencyLevel := event.msg.consistencyLevel
+				if !isEventConfirmed(consistencyLevel, blockEvents.header, now, height) {
+					logger.Debug("event not confirmed", zap.String("txId", event.TxId), zap.Uint8("consistencyLevel", consistencyLevel))
```
Comment (h0ngcha0) [inline node/pkg/alephium/watcher.go:403]: [Nit] Would it easier to read the log if we log inside the `isEventConfirmed`, so we know if it's not confirmed due to height or timestamp?

File: integration_test/src/alph.ts [hunk 3]
```diff
@@ -82,7 +82,7 @@ export async function createAlephium(): Promise<AlephiumBridgeChain> {
   const bridgeRewardRouterAddress = addressFromContractId(bridgeRewardRouterId)
   const sequence = new Sequence()
   const defaultArbiterFee = 0n
-  const defaultConfirmations = 1
+  const defaultConfirmations = 0
```
Comment (h0ngcha0) [inline integration_test/src/alph.ts:85]: Question: why do we need to set the the default confirmation to `0` here?
Could you please remind me on the reasoning behind using timestamp?

**Review by h0ngcha0 — COMMENTED** [review]
LGTM
Only Nit comments and questions.

File: node/pkg/alephium/watcher.go (hunk 2 above)
Comment (Lbqds) [reply node/pkg/alephium/watcher.go:403]: Sounds great 👍, will update.

File: integration_test/src/alph.ts (hunk 3 above)
Comment (h0ngcha0) [reply integration_test/src/alph.ts:85]: Thanks a lot for the great explanation :+1:

**Review by h0ngcha0 — APPROVED** [review]
LGTM

**Comment by Lbqds** [comment]
Close this PR because it has already been merged, but github didn't automatically close it.

**Pull request closed without merging**